src/Common/HostResolvePool.cpp

```diff
@@ -237,10 +254,22 @@ void HostResolver::updateImpl(Poco::Timestamp now, std::vector<Poco::Net::IPAddr
     }
 
     for (auto & rec : merged)
-        if (rec.failed && rec.fail_time < last_effective_resolve)
-            rec.failed = false;
+    {
+            if (!rec.failed)
+                continue;
+
+            /// Exponential increased time for each consecutive fail
+            auto banned_until = now - Poco::Timespan(history.totalMicroseconds() * (1ull << (rec.consecutive_fail_count - 1)));
```

Comment:
Here I use `history.totalMicroseconds()` instead `history.totalSeconds()` in order to work with times less than 1 second in ut-tests.